connector/src/main/java/org/geysermc/connector/network/session/cache/ChunkCache.java

```diff
@@ -56,57 +52,66 @@ public void addToCache(Column chunk) {
         if (!cache) {
             return;
         }
-        ChunkPosition position = new ChunkPosition(chunk.getX(), chunk.getZ());
-        if (chunk.getBiomeData() == null && chunks.containsKey(position)) {
-            Column newColumn = chunk;
-            chunk = chunks.get(position);
-            for (int i = 0; i < newColumn.getChunks().length; i++) {
-                if (newColumn.getChunks()[i] != null) {
-                    chunk.getChunks()[i] = newColumn.getChunks()[i];
+
+        long chunkPosition = ChunkPosition.toLong(chunk.getX(), chunk.getZ());
+        Column existingChunk;
+        if (chunk.getBiomeData() != null // Only consider merging columns if the new chunk isn't a full chunk
+            && (existingChunk = chunks.getOrDefault(chunkPosition, null)) != null) { // Column is already present in cache, we can merge with existing
```

Comment:
If there something more performant/better about using `chunks.getOrDefault(chunkPosition, null)) != null` rather than `chunks.containsKey(chunkPosition)`?

Reply:
Slightly more performant, it simply returns `null` rather than fetching `null` from `Long2ObjectOpenHashMap#defRetValue`, eliminating a useless `GETFIELD` opcode.